java/src/jmri/jmrix/mqtt/MqttConsist.java

```diff
@@ -0,0 +1,248 @@
+        log.debug("Add to consist address {} direction{}", LocoAddress, directionNormal);
+        if (consistType == CS_CONSIST) {
+            addToConsistList(LocoAddress, directionNormal);
+            if (active) publish();
+        } else {
+            log.error("Consist Type Not Supported");
+            notifyConsistListeners(LocoAddress, ConsistListener.NotImplemented);
+        }
+    }
+
+    /**
+     * Restore a Locomotive to Consist, but don't write to
+     * the command station.  This is used for restoring the consist
+     * from a file or adding a consist read from the command station.
+     *
+     * @param LocoAddress is the Locomotive address to add to the locomotive
+     * @param directionNormal is True if the locomotive is traveling
+     *        the same direction as the consist, or false otherwise.
+     */
+    @Override
+    public synchronized void restore(DccLocoAddress LocoAddress, boolean directionNormal) {
+        log.debug("Restore to advanced consist address {} direction {}", LocoAddress, directionNormal);
+
+        if (consistType == CS_CONSIST) {
+            addToConsistList(LocoAddress, directionNormal);
+        } else {
+            log.error("Consist Type Not Supported");
+            notifyConsistListeners(LocoAddress, ConsistListener.NotImplemented);
+        }
+    }
+
+    /**
+     * Remove a Locomotive from this Consist.
+     *
+     * @param LocoAddress is the Locomotive address to add to the locomotive
+     */
+    @Override
+    public synchronized void remove(DccLocoAddress LocoAddress) {
+        log.debug("Remove from advanced consist address {} direction {}", LocoAddress);
+
+        if (consistType == CS_CONSIST) {
+            removeFromConsistList(LocoAddress);
+            if (active) publish();
+        } else {
+            log.error("Consist Type Not Supported");
+            notifyConsistListeners(LocoAddress, ConsistListener.NotImplemented);
+        }
+    }
+
+
+    /**
+     * Activates the consist for use with a throttle
+     */
+    public void activate(){
+
+        log.info("Activating consist {}", consistID);
+        active = true;
+        publish();
+    }
+
+    /**
+     * Deactivates and removes the consist from a throttle
+     */
+     public void deactivate() {
+
+        log.info("Deactivating consist {}", consistID);
+        active = false;
+        // Clear MQTT message
+        jmri.util.ThreadingUtil.runOnLayoutEventually(() -> {
+            mqttAdapter.publish(this.sendTopicPrefix.replaceFirst("\\$address", String.valueOf(consistAddress.getNumber())), "");
+        });
+
+    }
+
+    private String getConsistMakeup() {
+
+        String consistMakeup = "";
+
+        for (DccLocoAddress  address : consistDir.keySet()) {
+            consistMakeup = consistMakeup.concat(consistDir.get(address) ? "":"-").concat(String.valueOf(address.getNumber())).concat(" ");
+        };
```

Comment:
Static analysis failes on this line due to the unnecessary semicolon.

Reply:
Fixed